units/sleep_scheduler.go

```diff
@@ -282,7 +299,21 @@ func (j *sleepSchedulerJob) makeStopAndStartJobs(ctx context.Context, _ evergree
 	hostIDsToStart := make([]string, 0, len(hostsToStart))
 	for i := range hostsToStart {
 		h := hostsToStart[i]
-		startJobs = append(startJobs, NewSpawnhostStartJob(&h, evergreen.ModifySpawnHostSleepSchedule, sleepScheduleUser, ts.Format(TSFormat)))
+		waitUntil := now.Add(time.Duration(i) * time.Second)
+		if waitUntil.After(latestWaitUntil) {
```

Comment:
Curious what is the actual rate limit AWS imposes? Just want to check that staggering each job by a second is enough to keep it at bay

Reply:
It's in [the docs](https://docs.aws.amazon.com/ec2/latest/devguide/ec2-api-throttling.html), but basically, there's multiple overlapping limits for each API call, and they restrict when any one of the limits is hit. For `StartInstances`, the limiting factor is that you make 5 requests in a burst, then you can make at most 2 requests/sec in a steady state. For `StopInstances`, you can make 200 in a single burst of requests and then 5 requests/sec in a steady state.

We also need to use `DescribeInstances` for each stop/start, which we use in a bunch of places, so that can also be rate limited.